framework/web/UrlRule.php

```diff
@@ -96,6 +96,14 @@ class UrlRule extends Object implements UrlRuleInterface
      * @since 2.0.10
      */
     public $normalizer;
+    /**
+     * @var bool Value used when pattern contains only optional params. If `true` rule pattern will contain slash at
+     * the end of param pattern (last optional param can not be omitted if all other patterns are not omitted), if `false`
+     * rule pattern will contain slash at the beginning of param pattern (first optional param can not be omitted
+     * if all other patterns are not omitted).
+     * @since 2.0.12
+     */
+    public $trailingSlashForDefaults = false;
```

Comment:
@samdark Why buggy? If I have pattern like `<language>/<category>/<slug>` when all part are optional, I may have want omit params from beginning, not from the end. With `$trailingSlashForDefaults = true` valid URLs will be:
* `en/my-default-category/project-slug`
* `my-default-category/project-slug`
* `project-slug`

With `$trailingSlashForDefaults = false` valid URLs will be:
* `en/my-default-category/project-slug`
* `en/my-default-category`
* `en`

Both cases seems valid to me.